Feature/team

makarman opened the pull request:
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=738)


## Code reviewers

- [x] @dmytrorevak 
- [ ] @IvanAndrukh
- [ ] @JjuliaP
- [x] @makarman
- [ ] @noobhunters
- [x] @Silent777
- [ ] @Staryj812

### Second Level Review

- [ ] @ssOleg
- [ ] @lhalam

Add Team app
    
    Added meat model
    Added migrations
    Added docstraings
    Changed settings

## CHECK LIST
- [x]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

lhalam commented on eventually/eventually/settings.py:
```diff
@@ -36,6 +36,7 @@
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
+    'team.apps.TeamConfig'
```
use app name `'team'`

lhalam commented on eventually/team/migrations/0001_initial.py:
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
```
Merge migrations file to one.
0001_initial.py
0002_auto_20171029_2302.py
0003_auto_20171030_1150.py
0004_auto_20171030_1155.py
0005_auto_20171030_1314.py

output: only 0001_initial.py file.

makarman replied:
done

lhalam commented on eventually/team/models.py:
```diff
@@ -0,0 +1,124 @@
+    name = models.CharField(max_length=30)
+    description = models.TextField(null=True)
+    image = models.CharField(max_length=300, null=True)
+    # owner_id= models.ForeignKey(User, on_delete=models.CASCADE)
```
delete commented code.
in the repository, it should not be.

makarman replied:
Done

lhalam commented on eventually/team/models.py:
```diff
@@ -0,0 +1,124 @@
+                    the certain team was last time edited.
+    """
+    name = models.CharField(max_length=30)
+    description = models.TextField(null=True)
```
Is the description field obligatory?

makarman replied:
Done

Silent777 commented on eventually/team/models.py:
```diff
@@ -0,0 +1,124 @@
+
+    """
+    .. class:: Team
+    Describing of team untity.
```
There is no word "untity"

redmyt commented on eventually/team/models.py:
```diff
@@ -0,0 +1,119 @@
+from django.db import models
+from django.db import IntegrityError
+
+# Create your models here.
```
Why do we need this line here?)

makarman replied:
We don't need it.

redmyt commented on eventually/team/models.py:
```diff
@@ -0,0 +1,119 @@
+        """
+        Static method that creates instance of Team class and creates databes
+        row with the accepted info.
+        :param name: team's name
```
add types for each parameter in the docstring

makarman replied:
done

lhalam commented on eventually/team/models.py:
```diff
@@ -0,0 +1,125 @@
+    def to_dict(self):
+        """
+        Method that converts team object to dictionary.
+        :return: dictionaty with team's information
```
add  empty line 
add example

makarman replied:
done

lhalam commented on eventually/team/models.py:
```diff
@@ -0,0 +1,125 @@
+            team.save()
+            return team
+        except IntegrityError:
+            return None
```
`pass`

makarman replied:
Done

lhalam commented on eventually/team/models.py:
```diff
@@ -0,0 +1,125 @@
+        try:
+            team.save()
+            return team
+        except IntegrityError:
```
Why is IntegrityError, and not ValueError?

makarman replied:
Done

Silent777 commented on eventually/team/models.py:
```diff
@@ -0,0 +1,169 @@
+class Team(models.Model):
+
+    """
+    .. class:: Team
```
![image](https://user-images.githubusercontent.com/33020662/32243841-4bee49aa-be7f-11e7-82d9-fee256f62901.png)
We'll get this as a result

makarman replied:
Done

Silent777 commented on eventually/team/models.py:
```diff
@@ -0,0 +1,169 @@
+        :type: create_date: datetime
+    """
+    name = models.CharField(max_length=30)
+    description = models.TextField(max_length=1024, blank=True)
```
If you specify a max_length attribute, it will be reflected in the Textarea widget of the auto-generated form field. However it is not enforced at the model or database level. Use a CharField for that

makarman replied:
Done

Silent777 replied:
don`t forget to make migrations

Silent777 commented on eventually/team/models.py:
```diff
@@ -0,0 +1,169 @@
+
+        :Example:
+
+        >>> from team.models import Team
```
?????

makarman replied:
Done.

Silent777 commented on eventually/team/models.py:
```diff
@@ -0,0 +1,169 @@
+
+        :return: team name, team description
+
+        :Example:
```
We do not need an example here

makarman replied:
Done.

Silent777 commented on eventually/team/models.py:
```diff
@@ -0,0 +1,169 @@
+        try:
+            team.save()
+            return team
+        except ValueError:
```
except (ValueError, IntegrityError) & import error from django.db

Silent777 commented on eventually/team/models.py:
```diff
@@ -0,0 +1,169 @@
+        try:
+            team.save()
+            return team
+        except ValueError:
```
except (ValueError, IntegrityError) & import error from django.db

makarman replied:
Done.

lhalam commented on eventually/team/models.py:
```diff
@@ -0,0 +1,163 @@
+            'name': self.name,
+            'description': self.description,
+            'image': self.image,
+            'created_at': self.created_at,
```
convert to timestamp

lhalam reviewed (dismissed):
mini fix and reduce the number of entries commits

makarman committed:
Checked

Add Team app
finish docstring
commanted view
fix problems with travis
Fix pull request
deleted commment and added types in docstring
added example
added id param
